src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+import java.util.Scanner;
+
 public class Duke {
+    public static void printTasks(Task[] tasks, int taskCount) {
+        System.out.println("Here are the tasks in your list:");
+        int taskNumber = 1;
+        for (int i = 0; i < taskCount; i++) {
+            System.out.print(taskNumber + ".");
+            tasks[i].printTask();
+            taskNumber++;
+        }
+    }
+
+    public static void markTask(Task[] tasks, int taskNumber) {
+        System.out.println("Nice! I've marked this task as done:");
+        tasks[taskNumber].markAsDone();
+        System.out.print(" ");
+        tasks[taskNumber].printTask();
+    }
+
+    public static void unmarkTask(Task[] tasks, int taskNumber) {
+        System.out.println("OK, I've marked this task as not done yet:");
+        tasks[taskNumber].markAsNotDone();
+        System.out.print(" ");
+        tasks[taskNumber].printTask();
+    }
+
+    public static void handleMarkAndUnmarkRequest(String userCommand,String[] arrayOfCommand,Task[] tasks) {
+        arrayOfCommand = userCommand.split(" ",2);
+        if (arrayOfCommand.length < 2 || arrayOfCommand[1].isEmpty()) {
+            System.out.print(Guide.MARK_AND_UNMARK_REQUEST_FORMAT);
+            throw new StringIndexOutOfBoundsException();
+        }
+        int taskNumber = Integer.parseInt(arrayOfCommand[1]);
+        if (arrayOfCommand[0].equals("mark")) {
+            markTask(tasks, taskNumber - 1);
+        } else {
+            unmarkTask(tasks, taskNumber - 1);
+        }
+    }
+
+    public static void handleTodoDeadlineAndEvent(String userCommand,int taskCount,String[] arrayOfCommand,Task[] tasks){
+        if (userCommand.startsWith("todo")) {
+            arrayOfCommand = userCommand.split(" ", 2);
+            if (arrayOfCommand.length < 2 || arrayOfCommand[1].isEmpty()) {
+                System.out.println(Guide.TODO_REQUEST_FORMAT);
+                throw new StringIndexOutOfBoundsException();
+            }
+            tasks[taskCount] = new Todo(arrayOfCommand[1]);
+        }
```

Comment:
Maybe can try to catch the error instead of exiting the program when encountering invalid inputs.